### PR TITLE
nsc-events-nextjs_7_360_creator-access

### LINKED ISSUE #360: Event Detail Page 1: Creator Access
As a developer, I want to make it so that:

- [x] Creators can only edit, delete, and archive their own events
- [x] Edit, Archive, and Delete buttons for events not created by creator user are hidden (these buttons should only be visible for their own events)
- [x] Messages should be updated to accurately portray success/error status of user's actions.

Related to #358

### PR DESCRIPTION
**Pull request by intisarosman1**
Resolves #360

This PR makes it so that creators are not able to edit, archive, or delete buttons for events not created by them. The buttons will only be visible if it is their own events.

I also made some small fixes for error handling.

An example of a creator user logged in attempting to delete an event created by another user.

https://github.com/SeattleColleges/nsc-events-nextjs/assets/77607212/017573e8-ca51-4f0d-8549-505dfa23189b

Related to #358

### CODE REVIEW

**Review by taylorpapke — APPROVED** [review]
Works as expected. Edit, archive and delete buttons are not visible when I go to another creators event while logged in as a creator. Will this feature also be available for admins with other admins events? @heosman

**Comment by kaimanasse15** [comment]
https://github.com/SeattleColleges/nsc-events-nextjs/assets/91504156/d4f53b90-3393-495c-a58e-b0e5c87f767d

It works as my expected. As a creator, I am unable to fix or change anything

**Review by Robel-003 — APPROVED** [review]
Working as expected and it's looks good. I'm only able to archive, edit, and delete my events only when signed in as a creator.

**Review by alasali1 — APPROVED** [review]
It looks like it is working as expected. I can only update, edit and delete my events.

File: app/event-detail/page.tsx [hunk 1]
```diff
@@ -30,13 +30,14 @@ interface SearchParams {
...
   const router = useRouter();
   const [event, setEvent] = useState(activityDatabase)
-  const [isAuthed, setAuthed] = useState(false);
   const [token, setToken] = useState("")
   const [dialogOpen, setDialogOpen] = useState(false);
   const [archiveDialogOpen, setArchiveDialogOpen] = useState(false);
   const [snackbarMessage, setSnackbarMessage] = useState("")
   const [editDialogOpen, setEditDialogOpen] = useState(false);
   const [attendDialogOpen, setAttendDialogOpen] = useState(false);
+  const [userId, setUserId] = useState("");
```
Comment (bcko) [inline app/event-detail/page.tsx:39]: feedback 1

File: app/event-detail/page.tsx [hunk 2]
```diff
@@ -187,7 +194,7 @@ const toggleArchiveDialog = () => {
         </Card>
           <div style={ { width: '100vh', display: 'flex' }}>
             <div style={ { display: 'flex', width: '100vh', gap: '25px',  justifyContent: 'center', alignItems: 'center', marginLeft: '13vh' } }>
-              {isAuthed && (
+              {(userRole === "admin" || (userRole === "creator" && event?.createdByUser === userId)) && (
```
Comment (bcko) [inline app/event-detail/page.tsx:197]: feedback2

File: components/ArchiveDialog.tsx [hunk 3]
```diff
@@ -29,11 +29,16 @@ const ArchiveDialog = ({ isOpen, eventId, dialogToggle }: ArchiveDialogProps) =>
                     'Authorization': `Bearer ${token}`
                 }
             });
-            return response.json();
-        } catch (error) {
-            console.error('error: ', error)
-        }
-    }
+            if (!response.ok) {
+                throw new Error(`Failed to archive event: ${response.statusText}`);
```
Comment (bcko) [inline components/ArchiveDialog.tsx:33]: feedback 3

**Pull request merged**